### PR TITLE
Documentation Update

### PR DESCRIPTION
**Pull request by fbaligand**
@purbon I have cleaned README.md. Are you OK with that ? 
This is exactly my version of README.md when you say "LGTM"

### CODE REVIEW

**Comment by fbaligand** [comment]
@purbon did you see this PR ?
Are you ok with that ?

**Comment by fbaligand** [comment]
@purbon @jsvd @ph @ycombinator @suyograo  @colinsurprenant @jordansissel 
Hi guys,
Can you review and merge my pull request ?
It only concerns plugin documentation.

**Comment by purbon** [comment]
Hi, will take care of it tomorrow. Sorry it took long to come back, got
into other important task and delayed this without noticing.

/purbon

**Comment by fbaligand** [comment]
Nice, thank you very much @purbon !

File: logstash-filter-aggregate.gemspec [hunk 1]
```diff
@@ -6,7 +6,7 @@ Gem::Specification.new do |s|
   s.description = "This gem is a logstash plugin required to be installed on top of the Logstash core pipeline using $LS_HOME/bin/plugin install gemname. This gem is not a stand-alone program"
   s.authors = ["Elastic", "Fabien Baligand"]
   s.email = 'info@elastic.co'
-  s.homepage = "http://www.elastic.co/guide/en/logstash/current/index.html"
+  s.homepage = "https://github.com/logstash-plugins/logstash-filter-aggregate"
```
Comment (purbon) [inline logstash-filter-aggregate.gemspec]: Hi, we actually point all plugins to use as homepage the logstash one, not sure if it makes complete sense, but being consistent makes sense. What do you think?

File: README.md [hunk 2]
```diff
@@ -106,18 +133,4 @@ Default value: `false`
 - **timeout:**  
 The amount of seconds after a task "end event" can be considered lost.  
 The task "map" is then evicted.  
-The default value is 0, which means no timeout so no auto eviction.
-
-## Need Help?
```
Comment (purbon) [inline README.md]: If moving this to the ```BUID.md```, can we add a link to that document in this README? like use users could find it as README is the first thing they see.

File: logstash-filter-aggregate.gemspec (hunk 1 above)
Comment (fbaligand) [inline logstash-filter-aggregate.gemspec]: @purbon 
I understand the relevance to have a centralized documentation.
But for aggregate plugin, one thing is currently missing : There is no entry "aggregate" in the filter plugins list (https://www.elastic.co/guide/en/logstash/current/filter-plugins.html).
Can it be added ?
In this pull request, I have enhanced asciidoc documentation for that.

File: README.md (hunk 2 above)
Comment (fbaligand) [inline README.md]: After thinking a little about that, I realize that "Need Help" is more a "plugin user" thing than a "plugin developer" thing.
=> I get it back to README.md
Tell me what you think about that.

File: logstash-filter-aggregate.gemspec (hunk 1 above)
Comment (purbon) [inline logstash-filter-aggregate.gemspec]: Well, after giving a bit of thought I think it makes sense to use this. We also open an issue to decide what to do with this, as you noticed, is not consistent. ref: https://github.com/elastic/logstash/issues/3566

File: README.md [hunk 3]
```diff
@@ -106,18 +133,9 @@ Default value: `false`
 - **timeout:**  
 The amount of seconds after a task "end event" can be considered lost.  
 The task "map" is then evicted.  
-The default value is 0, which means no timeout so no auto eviction.
+The default value is 0, which means no timeout so no auto eviction.  
+
 
 ## Need Help?
 
 Need help? Try #logstash on freenode IRC or the https://discuss.elastic.co/c/logstash discussion forum.
```
Comment (purbon) [inline README.md]: I like that, althought I think is also nice to have a short/small pointer to the build.md document, some sentence like, if you like to contribute, or build the plugin, see this.

What do you think?

**Comment by purbon** [comment]
@fbaligand I just left you an small comment about having a link from the README to the BUILD document, if we resolve than then I'm very good to go.

Thanks for your patience and effort man!

File: README.md (hunk 3 above)
Comment (fbaligand) [inline README.md]: Okay, I will add a link to BUILD.md in README.md.
I will do it tonight.

File: logstash-filter-aggregate.gemspec (hunk 1 above)
Comment (fbaligand) [inline logstash-filter-aggregate.gemspec]: @purbon 
OK, I keep https://github.com/logstash-plugins/logstash-filter-aggregate as home page.
But I'm still strongly interested to add aggregate plugin here : https://www.elastic.co/guide/en/logstash/current/filter-plugins.html

Is it possible ? Does it require something particular to be published here ?

**Comment by fbaligand** [comment]
@purbon 
That's done !

Furthermore, I don't know if you saw, but I have asked a question about how to publish aggregate documentation to official logstash documentation.

**Comment by purbon** [comment]
@fbaligand nowadays the documentation is generated with the plugins bundled with the logstash distribution. I think a good improvement for the documentation would be to have also there all plugins hosted in the logstash-plugins organization, might be in a collateral page.

Can you open an issue for this in the logstash repo?

Thanks man!

- purbon

**Comment by purbon** [comment]
LGTM

**Comment by fbaligand** [comment]
@purbon 
Thanks for the merge !

Concerning the integration in logstash documentation ; it's not as simple as that.
For example : the "elapsed" plugin is a "community-driven plugin", which is not in official logstash distrib.
However, "elapsed" plugin is integrated in logstash documentation :
https://www.elastic.co/guide/en/logstash/current/plugins-filters-elapsed.html
With a particular note :
`This is a community-maintained plugin! It does not ship with Logstash by default, but it is easy to install by running bin/plugin install logstash-filter-elapsed.`

**Comment by purbon** [comment]
@fbaligand I would suggest to open an issue in the main project, this would help decide and fix the situation. Makes sense?

**Comment by fbaligand** [comment]
@purbon 
OK, I will create an issue.
To be clear, my issue is "integrate aggregate documentation", or more globally, "integrate community-driven plugins documentation" ?

**Comment by purbon** [comment]
Good!!, I guess is the last one, you aim to have plugins like yours integrated, isn't?

**Comment by fbaligand** [comment]
@purbon 
OK, I will open a global issue for all logstash plugins which are hosted in "logstash-plugins" github account.
It makes sense to open a global issue, regarding the expected benefits announced here :
https://www.elastic.co/guide/en/logstash/current/_how_to_write_a_logstash_filter_plugin.html#_contributing_your_source_code_to_ulink_url_https_github_com_logstash_plugins_logstash_plugins_ulink_3

**Comment by purbon** [comment]
please share your concerns, feedback is more than appreciate @fbaligand, and again sorry it took me long time to come back here :-(

**Comment by fbaligand** [comment]
No problem, I'm happy that my pull request is finally merged :)
For information, I have published the plugin on rubygems.org. The number of downloads is encouraging !
Now, next step : having aggregate documentation integrated in official documentation :)

**Comment by purbon** [comment]
go for it :+1 !! here to help.

BTW, still not sure how the rubygems.org downloads work? something numbers look strange to me. Do you know what do they target specifically ?

**Comment by fbaligand** [comment]
I have no idea !
But I agree it sounds strange : once I published aggregate plugin, I had 4 downloads in the following minute !

**Comment by fbaligand** [comment]
@purbon
The issue is now open !
https://github.com/elastic/logstash/issues/3586

**Pull request closed without merging**